rusk-wallet/src/wallet/file.rs

```diff
@@ -4,107 +4,155 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-use std::fmt;
+use std::fmt::{Debug, Display};
+use std::hash::Hash;
 use std::path::{Path, PathBuf};
 use std::str::FromStr;
 
-/// Provides access to a secure wallet file
-pub trait SecureWalletFile {
-    /// Returns the path
-    fn path(&self) -> &WalletPath;
-    /// Returns the hashed password
-    fn pwd(&self) -> &[u8];
+use crate::dat::DatFileVersion;
+use crate::{Error, SecureWalletFile};
+
+use super::file_service::WalletFilePath;
+
+/// Wallet file structure that contains the path of the wallet file, the hashed
+/// password, and the file version
+#[derive(Debug, Clone)]
+pub struct WalletFile {
+    path: WalletPath,
+    pwd: Vec<u8>,
```

Comment:
Saving the password here in the memory is necessary? I don't think its secure also given this struct is gonna live for the duration of the wallet's instance. Can we have it like before where the password wasn't stored in memory?

Reply:
I didn't touch the `WalletFile` shape and it had the `pwd` property before refactoring.